Source/Dafny/Resolver.cs

```diff
@@ -11932,6 +11949,42 @@ private class RBranchExpr : RBranch {
           "Internal Error: Unknown failure type in :- statement");
         return;
       }
+
+      Expression lhsExtract = null;
+      if (expectExtract) {
+        Method caller = codeContext as Method;
+        if (caller != null && caller.Outs.Count == 0 && s.ExpectToken == null) {
+          reporter.Error(MessageSource.Resolver, s.Rhs.tok, "Expected {0} to have a Success/Failure output value",
+            caller.Name);
+          return;
+        }
+
+        lhsExtract = s.Lhss[0];
+        if (lhsExtract is ExprDotName lex) {
```

Comment:
Have the `s.Lhss` expressions been resolved by this point? If so, each `s.Lhss[i]` should be one of:
* `IdentifierExpr`
* `MemberSelectExpr` (not `ExprDotName`)
* `SeqSelectExpr` (with `.E0 != null` and `.E1 == null`)
* `MultiSelectExpr`

Reply:
Improved